### PR TITLE
Add search modal

### PR DESCRIPTION
**Pull request by chdsbd**

### CODE REVIEW

File: frontend/src/components/SearchModal.jsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+  handleSearch = (event) => {
+    if (event.key === 'Enter') {
+      this.search(this.state.query)
+    }
+  }
+
+  handleKeyPress = (event) => {
+    if (document.activeElement.tagName !== 'BODY') return
+    const pressF = event.key === 'f' && !event.ctrlKey && !event.meta
```
Comment (sbdchd) [inline frontend/src/components/SearchModal.jsx]: What about shift?

File: frontend/src/components/SearchModal.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  componentDidUpdate (_prevProps, prevState) {
+    if (prevState.query !== this.state.query) {
+      this.search(this.state.query)
+    }
+  }
+
+  render () {
+    const { searchResults, loading } = this.props
+    let content = <p className="text-muted">Search all recipes (ingredients, steps, and metadata).</p>
```
Comment (sbdchd) [inline frontend/src/components/SearchModal.jsx]: ternary?

File: frontend/src/components/scss/search.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
+.search--results {
+  overflow-y: auto;
+  max-height: calc(60vh - 125.5px);
```
Comment (sbdchd) [inline frontend/src/components/scss/search.scss]: comment

File: frontend/src/components/scss/search.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  padding: 1px;
+}
+
+.search--position-top {
+  height: calc(100vh - 160px);
```
Comment (sbdchd) [inline frontend/src/components/scss/search.scss]: comment

File: frontend/src/store/actions.js [hunk 5]
```diff
@@ -667,6 +670,62 @@ export const fetchRecipeList = (teamID) => dispatch => {
...
+export const setSearchResults = (results) => ({
+  type: SET_SEARCH_RESULTS,
+  results,
+})
+
+export const clearSearchResults = () => ({
+  type: CLEAR_SEARCH_RESULTS,
+})
+
+export const incrLoadingSearch = (val) => {
```
Comment (sbdchd) [inline frontend/src/store/actions.js]: make a `incrLoadingSearch` and a `decrLoadingSeach` that take no arguments

File: frontend/src/components/RecipeItem.jsx [hunk 6]
```diff
@@ -33,8 +33,7 @@ function collect (connect, monitor) {
   }
 }
 
-@DragSource(DragDrop.RECIPE, recipeSource, collect)
-export default class RecipeItem extends React.Component {
+class RecipeItem extends React.Component {
```
Comment (sbdchd) [inline frontend/src/components/RecipeItem.jsx]: why not just `export` & remove the export at the bottom?

File: frontend/src/components/SearchModal.jsx (hunk 1 above)
Comment (chdsbd) [reply frontend/src/components/SearchModal.jsx]: that's covered by `event.key`. If we wanted `shift-f`, we'd check for `F`, not `f`

File: frontend/src/components/SearchModal.jsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    if (event.key === 'Enter') {
+      this.search(this.state.query)
+    }
+  }
+
+  handleKeyPress = (event) => {
+    if (document.activeElement.tagName !== 'BODY') return
+    const pressF = event.key === 'f' && !event.ctrlKey && !event.meta
+    if (pressF) {
+      this.setState({ show: true })
```
Comment (sbdchd) [inline frontend/src/components/SearchModal.jsx]: shouldn't below be in a call back?

**Pull request merged**